packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollView.java

```diff
@@ -1109,6 +1109,54 @@ public void setContentOffset(ReadableMap value) {
     }
   }
 
+  public void setContentInset(ReadableMap insets) {
+    int left = 0;
+    int top = 0;
+    int right = 0;
+    int bottom = 0;
+    if (insets != null) {
+      if (insets.hasKey("left")) {
+        left = (int) PixelUtil.toPixelFromDIP(insets.getDouble("left"));
+      }
+      if (insets.hasKey("top")) {
+        top = (int) PixelUtil.toPixelFromDIP(insets.getDouble("top"));
+      }
+      if (insets.hasKey("right")) {
+        right = (int) PixelUtil.toPixelFromDIP(insets.getDouble("right"));
+      }
+      if (insets.hasKey("bottom")) {
+        bottom = (int) PixelUtil.toPixelFromDIP(insets.getDouble("bottom"));
+      }
+    }
+
+    setPadding(left, top, right, bottom);
+    // simulate iOS behavior
+    setClipToPadding(false);
+
+    // If clipping of subviews is enabled, update the clipping rect
+    if (mRemoveClippedSubviews) {
+      updateClippingRect();
+    }
+
+    // Force a layout pass to ensure the new padding is applied.
+    requestLayout();
+    invalidate();
+
+    // Post a runnable to adjust the scroll position if needed.
+    // (Using post() ensures this happens after the layout pass.)
+    post(new Runnable() {
+      @Override
+      public void run() {
+        int maxScrollY = getMaxScrollY();
+        // If the current scroll offset is now beyond the new maximum,
+        // adjust it to the new maximum.
+        if (getScrollY() > maxScrollY) {
+          scrollTo(getScrollX(), maxScrollY);
+        }
+      }
+    });
```

Comment:
would this show a flicker or an animation or does it happen immediately? 🤔

Reply:
I'd say it happens immediately. Basically it fixes a very strange problem. Without this code problems looks like this:

https://github.com/user-attachments/assets/30e57fa1-7c0c-4573-afc2-df4a762210eb

Even though I can clearly see that setter gets called:

<img width="844" alt="image" src="https://github.com/user-attachments/assets/6960c1db-15ff-4da5-844a-52ef621d3c40" />

There is still a frozen area of the `contentInset`. But it disappear as soon as I start scrolling. So I thought to add `scrollTo` programmatically. Maybe there is a better fix for this problem - let me know if you find one 🙌